OpenRA.Mods.Common/Lint/CheckCursors.cs

```diff
@@ -0,0 +1,54 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Lint
+{
+	class CheckCursors : ILintMapPass
+	{
+		public void Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Map map)
+		{
+			var fileSystem = modData.DefaultFileSystem;
+			var sequenceYaml = MiniYaml.Merge(modData.Manifest.Cursors.Select(s => MiniYaml.FromStream(fileSystem.Open(s), s)));
+			var nodesDict = new MiniYaml(null, sequenceYaml).ToDictionary();
+
+			var cursors = new List<string>();
+			foreach (var s in nodesDict["Cursors"].Nodes)
+				foreach (var sequence in s.Value.Nodes)
+					cursors.Add(sequence.Key);
+
+			var rules = map.Rules;
+			foreach (var actorInfo in rules.Actors)
```

Comment:
FYI: this doesn't account for the custom terrain move cursors defined in tilesets, but this should be fixed in a followup PR (which I may take on after this one has been merged) by changing `MobileInfo.Cursor` to support tileset-specific overrides.